glide-core/src/client/value_conversion.rs

```diff
@@ -178,6 +179,42 @@ pub(crate) fn convert_to_expected_type(
             _ => Err((
                 ErrorKind::TypeError,
                 "Response couldn't be converted to an array of doubles",
+            format!("(response was {:?})", value),
+            )
+            .into()),
+        },
+        // command returns an array of 2 elements, where second is a nested array of 1+ elements which are arrays of 2 elements
+        // we convert the nested array to a map as we do in `MapOfStringToDouble`
+        /*
+        127.0.0.1:6379> zmpop 1 z1 min count 10
+        1) "z1"
+        2) 1) 1) "2"
+              2) (double) 2
+           2) 1) "3"
+              2) (double) 3
+         */
+        ExpectedReturnType::ZMPopReturnType => match value {
+            // command returns nil or
+            Value::Nil => Ok(value),
+            // array of 2 elements, where second is a nested array
+            Value::Array(array)
+                if array.len() == 2 && matches!(array.get(1).unwrap(), Value::Array(_)) =>
```

Comment:
If you're going to unwrap anyway, you can just use array indexing syntax.
```suggestion
                if array.len() == 2 && matches!(array[1], Value::Array(_)) =>
```